var/spack/repos/builtin/packages/aocc/package.py

```diff
@@ -0,0 +1,89 @@
+    multi-threading and processor support that includes global optimization,
+    vectorization, inter-procedural analyses, loop transformations,
+    and code generation.
+    AMD also provides highly optimized libraries,
+    which extract the optimal performance from
+    each x86 processor core when utilized.
+    The AOCC Compiler Suite simplifies and accelerates development and
+    tuning for x86 applications.
+    Please install only if you agree to terms and conditions depicted
+    under : http://developer.amd.com/wordpress/media/files/AOCC_EULA.pdf
+    Example for installation: \'spack install aocc +license-agreed\'
+    '''
+    family = 'compiler'
+    AOCC_HOME = None
+    homepage = "https://developer.amd.com/amd-aocc/"
+    key_part_1 = '500940ce36c19297dfba3aa56dcef33b'
+    key_part_2 = '6145867a1f34890945172ac2be83b286'
+
+    site_nm = 'developer.amd.com'
+    tar_nm = 'aocc-compiler-2.2.0.tar'
+    version(ver="2.2.0",
+            sha256='{0}{1}'.format(key_part_1, key_part_2),
+            url='http://{0}/wordpress/media/files/{1}'.format(site_nm, tar_nm))
+
+    # Licensing
+    lic_nm = 'AOCC_EULA.pdf'
+    license_required = True
+    license_comment = '#'
+    license_files = [lic_nm]
+    license_url = 'http://{0}/wordpress/media/files/{1}'.format(site_nm, lic_nm)
+    install_example = "spack install aocc +license-agreed"
+
+    depends_on('libxml2')
+    depends_on('zlib')
+    depends_on('ncurses')
+    depends_on('libtool')
+    depends_on('texinfo')
+
+    variant('license-agreed', default=False,
+            description='Agree to terms and conditions depicted under : {0}'
+            .format(license_url))
+
+    @run_before('install')
+    def abort_without_license_agreed(self):
+        site_nm = 'developer.amd.com/wordpress/media'
+        lic_nm = 'AOCC_EULA.pdf'
+        license_url = 'http://{0}/files/{1}'.format(site_nm, lic_nm)
+        install_example = "spack install aocc +license-agreed"
+        if not self.spec.variants['license-agreed'].value:
+            raise InstallError("\n\n\nNOTE:\nUse +license-agreed " +
+                               "during installation " +
+                               "to accept terms and conditions " +
+                               "depicted under following link \n" +
+                               " {0}\n".format(license_url) +
+                               "Example: \'{0}\' \n".format(install_example))
+
+    def install(self, spec, prefix):
+        print("Installing AOCC Compiler ... ")
+        install_tree('.', prefix)
+
+    def setup_run_environment(self, env):
+        env.set('CC', join_path(self.spec.prefix.bin, 'clang'))
+        env.set('CXX', join_path(self.spec.prefix.bin, 'clang++'))
+        env.set('FC', join_path(self.spec.prefix.bin, 'flang'))
+        env.set('F77', join_path(self.spec.prefix.bin, 'flang'))
+        env.set('F95', join_path(self.spec.prefix.bin, 'flang'))
+
+        env.prepend_path('PATH', self.spec.prefix.bin)
```

Comment:
Spack will add `prefix.bin` to `PATH` by default so you don't need to set this explicitly.

Spack does not set `C_INCLUDE_PATH` or `CPLUS_INCLUDE_PATH` but it does set `CPATH`; therefore, if `aocc` is like `gcc` then based on https://gcc.gnu.org/onlinedocs/cpp/Environment-Variables.html, you also don't need to set those (in particular because you set `prefix.include` for both of them - it would be different if there were separate include directories for C and C++).

Reply:
Yes, I have removed these settings in my new PR https://github.com/spack/spack/pull/19345